app/views/shared/_alert.html.erb

```diff
@@ -1,5 +1,14 @@
+<%#
+locals:
+* type: One of: "info", "success", "warning", "error", "other". Defaults to "info".
+* text_tag: Optional override HTML tag for text content. Defaults to `p`.
+* class: Additional class names to add to alert wrapper.
+* message: Text to display in alert. Text can also be provided through block.
+%>
 <%
-  type = local_assigns.fetch(:type, 'other')
+  type = local_assigns.fetch(:type, 'info')
+  type = 'error' if type === 'alert'
+  type = 'info' unless ['info', 'success', 'warning', 'error', 'other'].include?(type)
```

Comment:
I would make this error in a much crankier way and let's fix our call sites that use now-old styles?

```suggestion
  type = local_assigns.fetch(:type, 'info')
  raise "unknown flash type=#{type}" unless %w[info success warning error other].include?(type)
```

Reply:
> I would make this error in a much crankier way and let's fix our call sites that use now-old styles?

We can update our own usages, but as [mentioned in another comment](https://github.com/18F/identity-idp/pull/4375#discussion_r516969034), some of these are coming from Devise, and don't appear to be customizable.

What might be possible is to try to normalize this somewhere else before it reaches the alert partial. `_flashes.html.erb` is an option, though ideally something before the view is rendered.

Reply:
> I would make this error in a much crankier way and let's fix our call sites that use now-old styles?

Raising error in 69df36977.

Updated existing usage in 035316bbc.

To handle the Devise case, 76daf5e8f adds a mapping of flash keys to their normalized form. This happens outside the `alert` partial, instead in `flashes`.

A few alternatives explored:

- This has apparently been discussed at length in the Devise project (https://github.com/heartcombo/devise/issues/3748, https://github.com/plataformatec/devise/issues/2051, https://github.com/plataformatec/devise/issues/2282, https://github.com/plataformatec/devise/pull/748). The closest thing to a workaround to try to control these as they're assigned is [this approach](https://github.com/heartcombo/devise/issues/2282#issuecomment-13636842), though it felt a bit overkill for our purposes.
- I was hoping for a slightly cleaner implementation of `_flashes` using `transform_keys`, but it apparently gets stuck in an infinite loop. Not entirely clear why.

```rb
<%
flash.
  to_hash.
  transform_keys { |key| ApplicationController::FLASH_KEY_MAPPING.fetch(key, key)
  slice(*ApplicationController::FLASH_KEYS).
  each do |type, message| %>
  <% if message.present? %>
    <%= render 'shared/alert', {
      type: type,
      message: safe_join([message.html_safe]),
      class: 'margin-bottom-4',
    } %>
  <% end %>
<% end %>
```